### PR TITLE
Fix oddities with generic cooked offal

### PR DESCRIPTION
**Pull request by chaosvolt**
#### Summary

SUMMARY: Bugfixes "Fix cooked offal recipes not needing heat to cook, make fried offal actually flavored as such"

#### Purpose of change

So I was gonna make this PR just a bundle of some misc easy fixes that barely warranted a separate PR each, but looking into the issue of cooked offal not using heat to cook actually turned out a lot more complicated than I expected, due to https://github.com/cataclysmbnteam/Cataclysm-BN/pull/1371 seeming to intend to reflavor generic cooked offal as fried offal, warranting more involved changes beyond just fixing the bugged recipe.

#### Describe the solution

1. Reflavored cooked offal as fried offal. This seems to be what the change to recipe and mention of "you can fry them" in the PR OP was talking about, which makes sense because just plain campfire generic offal no longer has any reason to exist now that every offal item has its own cooked variant. Positive enjoyment but much lower than the other more advanced fried offal items, same level of unhealthiness (it's literally just meat and oil). Also moved it to meat dishes since it's no longer just generic cooked item.
2. Fixed cooked/fried offal not having `surface_heat` for its cooking. Used same amount as most other fried meat recipes.
3. Per feedback from Coolthulhu, obsoleted the separate cooked offal item in favor of making it so you have to fry it to make it palatable. Fried offal also gets different name based off components, includes cannibal name in case we add human organs.

#### Describe alternatives you've considered

Honestly, it's been so long since generic non-mutant offal was functionally dummied out via never spawning anywhere and not being obtainable via harvest that we could likely remove it from recipes too, the odds of a save still have raw generic offal kicking around are fairly low.

#### Testing

Checked affected files for syntax and lint errors.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Adding oil requirement was meant mostly to make it a bit harder to cook. Yeah, surface heat was missing.
I'd like to make offal generic again, with only useful organs (brain?) being split out of it.

As for separate mutant recipe: I guess it could be removed. I think I added it out of "mental copy+paste". But if it stays, it should keep the oil, it has COOK 2 for it.

**Comment by chaosvolt** [comment]
> As for separate mutant recipe: I guess it could be removed. I think I added it out of "mental copy+paste". But if it stays, it should keep the oil, it has COOK 2 for it.

Ah right, didn't spot that it requires cooking 2. Would seem a bit weird to have cooked mutant offal be the only "cooked organ meat" item that requires more effort to cook admittedly, and keeping it as a "frying this is just so you can get it to the same bare minimum level of repeatability since it's even nastier" item means that either the restored cooked offal recipe should probably be obsoleted again, or it shouldn't allow mutant offal, presumably?

**Comment by Coolthulhu** [comment]
> Would seem a bit weird to have cooked mutant offal be the only "cooked organ meat" item that requires more effort to cook admittedly

It's alien shit, I ain't gotta explain shit.
But yeah, if fried offal covers it, the new recipe can be removed and fried offal should just get one of those special names when it is mutant.

**Comment by chaosvolt** [comment]
> But yeah, if fried offal covers it, the new recipe can be removed and fried offal should just get one of those special names when it is mutant.

I mean, being able to cook mutant offal the regular way separately while being able to fry it so it tastes better also works, I'm just not sure what to go with here because there are several conflicting approaches I could use here.

**Comment by chaosvolt** [comment]
Went ahead with that idea of axing cooked mutant offal and added name variations to the generic fried variant.

File: data/json/items/comestibles/carnivore.json [hunk 1]
```diff
@@ -279,7 +279,7 @@
     "type": "COMESTIBLE",
     "name": "raw mutant offal",
     "looks_like": "offal",
-    "description": "Unidentified organs of a mutated animal.  Wildly different between individual specimens, even those otherwise identical.  Many resemble unmutated organs on the outside, but their placement often clearly implies different function.  Weirdly colored fluids and the chemical smell make them unappetizing.",
+    "description": "Unidentified organs of a mutated animal.  Wildly different between individual specimens, even those otherwise identical.  Many resemble unmutated organs on the outside, but their placement often clearly implies different function.  Weirdly colored fluids and the chemical smell make them unappetizing.  Probably much better fried.",
```
Comment (Coolthulhu) [inline data/json/items/comestibles/carnivore.json:282]: I'd drop the extra comment, the UI will mention the recipe anyway.

**Pull request merged**